javacutil/src/main/java/org/checkerframework/javacutil/AnnotationUtils.java

```diff
@@ -334,26 +334,46 @@ public static boolean containsSameByName(
         return null;
     }
 
-    private static final Comparator<AnnotationMirror> ANNOTATION_ORDERING =
-            new Comparator<AnnotationMirror>() {
-                @Override
-                public int compare(AnnotationMirror a1, AnnotationMirror a2) {
-                    // AnnotationMirror.toString() prints the elements of an annotation in the
-                    // order in which they were written. So, use areSame to check for equality.
-                    if (AnnotationUtils.areSame(a1, a2)) {
-                        return 0;
-                    }
-
-                    String n1 = a1.toString();
-                    String n2 = a2.toString();
-
-                    // Because the AnnotationMirror.toString prints the annotation as it appears
-                    // in source code, the order in which annotations of the same class are
-                    // sorted may be confusing.  For example, it might order
-                    // @IntRange(from=1, to=MAX) before @IntRange(to=MAX,from=0).
-                    return n1.compareTo(n2);
-                }
-            };
+    /**
+     * Compares two annotation classes by lexicographically comparing their canonical names.
+     *
+     * @param a1 the first annotation class
+     * @param a2 the second annotation class
+     * @return the value {@code 0} if the name of the first class is equal to the name of the
+     *     second; a value less than {@code 0} if the name of the first class is lexicographically
+     *     less than the name of the second; and a value greater than {@code 0} if this name of the
+     *     first class is lexicographically greater than the name of the second.
+     */
+    public static int compareAnnotationClasses(
+            Class<? extends Annotation> a1, Class<? extends Annotation> a2) {
+        return a1.getCanonicalName().compareTo(a2.getCanonicalName());
+    }
+
+    /**
+     * Provide ordering for {@link AnnotationMirror} based on their fully qualified name. The
+     * ordering ignores annotation values when ordering.
+     *
+     * <p>The ordering is meant to be used as {@link TreeSet} or {@link TreeMap} ordering. A {@link
+     * Set} should not contain two annotations that only differ in values.
+     *
+     * @return an ordering over AnnotationMirrors based on their name
+     */
+    public static int compareAnnotationMirrors(AnnotationMirror a1, AnnotationMirror a2) {
```

Comment:
The documentation of this method, copied from `annotationOrdering` does not match its implementation. The ordering does not ignore annotation values.  I'm not sure if I should change the documentation or the implementation.

Reply:
Code comments indicate:

1. `annotationOrdering` assumes that no set contains two annotations that differ only in their values.  Is this a good assumption, given that we have made more annotations (even some type annotations) repeatable?

2. The implementation of ANNOTATION_ORDERING is broken when an annotation has values.

I would be inclined to make ANNOTATION_ORDERING respect values, and not use toString but compare the number of values/elements, and if they are equal to sort and compare them.

Reply:
RE 1. It's a good assumption if comparing AnnotationMirrors in sets in AnnotatedTypeMirrors, but I still think this method should work with annotations of the same class.

> I would be inclined to make ANNOTATION_ORDERING respect values, and not use toString but compare the number of values/elements, and if they are equal to sort and compare them.

I agree with this and will make that change in a separate PR.

Reply:
See #2962.